### ISSUE TITLE
v6rc3 allowOutsideOutDir and differents assets[] config for serve and build

### ISSUE THREAD

**Issue by istiti** [angular/angular-cli#10284]
### Versions
```
Angular CLI: 6.0.0-rc.3
Node: 8.11.1
OS: win32 x64
Angular: 6.0.0-rc.3
... animations, cli, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cdk: 6.0.0-rc.1
@angular/material: 6.0.0-rc.1
@angular-devkit/architect: 0.5.5
@angular-devkit/build-angular: 0.5.5
@angular-devkit/build-optimizer: 0.5.5
@angular-devkit/core: 0.5.5
@angular-devkit/schematics: 0.5.5
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 6.0.0-rc.3
@schematics/angular: 0.5.5
@schematics/update: 0.5.5
typescript: 2.7.2
webpack: 4.5.0
```

### Observed behavior
my option in old angular-cli.json file on assets[] `"allowOutsideOutDir":true` not working 

In old file I had two apps with different assets configuration and for ng serve I select appA and for ng build I select appB

this is my new angular.json I tried to add two different assets configuration in in my projects.iii.architect{} build and serve:
https://gist.github.com/istiti/5ce4aa0ac45d0be06c79f417b59a6281

I get: 
![image](https://user-images.githubusercontent.com/2789203/38622939-ab9916f6-3da4-11e8-9a4e-b544251674c8.png)

What is problem ? thanks 

ps: my old config https://gist.github.com/istiti/d702f613bd4ec4cb71f5ea131cbd579a

and I was running them using : 
`node --max-old-space-size=8192 ./node_modules/@angular/cli/bin/ng build --prod --app=prod --output-path dist/fr --locale fr --bh /fr/ --i18n-format xtb --i18n-file src/locale/fr.xtb`

**Comment by hansl** [comment]
Fixed in #10245.

**Comment by istiti** [comment]
@hansl i want avoid create new issue I tried extract my i18 tags using
`ng xi18n i10 --i18n-format xmb --output-path locale` it seems it works because file exist in src/local/ folder 

Then I tried to build my 1k files project using 
`ng build myProject--prod  --i18n-format xtb --i18n-file src/locale/fr.xtb`

I get 

> ERROR in The translation file (C:\Users\elvis\WebstormProjects\i10\src\locale\fr.xtb) locale must be provided. 

While file exist:
![image](https://user-images.githubusercontent.com/2789203/38658388-33d05dc4-3e25-11e8-9f88-f3c5afd95a0f.png)


Maybe command to build is wrong and need specify these arguments inside angular.json>projects.myProject.architect.build
